addons/rose/addon/components/rose/loading-button/index.js

```diff
@@ -0,0 +1,20 @@
+import button from '../button/index';
+
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class LoadingButton extends button {
+  // =actions
+  @tracked isLoading = false;
+
+  @action
+  async toggleRefresh() {
+    this.isLoading = true;
+    try {
+      await this.args.onClick();
+      this.isLoading = false;
+    } catch (e) {
+      throw new Error('error refreshing sessions');
```

Comment:
This button could be used to refresh any resource.  I'm also not sure if we should throw an error or just allow errors to occur.  Can we test what happens when network errors occur?

Reply:
Good point! I updated the `catch` to log the errors instead of throwing it as it made the state `isLoading` stuck and didn't let me refresh the sessions. This error(`transition aborted`) occurs when I repeatedly hit the refresh button. Now, when the error occurs, it logs to the console and the refresh functionality works as expected. Let me know if there's a better way to handle this!